src/posix/platform/uart.cpp

```diff
@@ -238,7 +238,7 @@ static void InitializeSessionSocket(void)
     // to simply ignore it.
 #if defined(SO_NOSIGPIPE)
     rval = 1;
-    VerifyOrExit((rval = setsockopt(sSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &rval, sizeof(rval)) != -1, OT_NOOP));
+    VerifyOrExit((rval = setsockopt(sSessionSocket, SOL_SOCKET, SO_NOSIGPIPE, &rval, sizeof(rval))) != -1, OT_NOOP);
```

Comment:
Nice find. This s tricky bug (do we not get compile error on this?) 👍 

How about breaking the statement into two lines?
- One line setting the `rval = setsockopt(...)` and 
- The other just a simple `VerifyOrExit(rval != -1, OT_NOOP)`.

I feel makes it more readable.

(also do we need the `rval = 1;` before this. Can we remove that as well?)

Reply:
@abtink 

> Nice find. This is a tricky bug (do we not get compile error on this?) +1

This actually will give a compiling error when building posix targets with `-DOT_DAEMON=ON`, added a build check for posix daemon.

>How about breaking the statement into two lines?
One line setting the rval = setsockopt(...) and
The other just a simple VerifyOrExit(rval != -1, OT_NOOP).

Changed, please review.